org.eclipse.lyo.oslc4j.codegenerator/src/org/eclipse/lyo/oslc4j/codegenerator/jsp/generateResourceShapeJsp.mtl

```diff
@@ -103,11 +103,11 @@ To revert to the default generated content, delete all content in this file, and
 
                     <h2>Describes</h2>
                     <ul>
-                        <%for(URI next : aResourceShape.getDescribes()) {%>
-                            <li>
-                                <a href="<%=next%>">
-                                    <%= next.toString().split("#")['[1]'/]%></a>
-                            </li>
+                        <%for(URI next : aResourceShape.getDescribes()) {
+                            String['[]'/] split = next.toString().split("['[#/]'/]+");
```

Comment:
right! That's what I tried to explain in the comment above. This is the best I can do, since "describes" is simply a URI. The best I can do is to try to split it, by finding the last spot where a "/" or "#" was used.  
At this stage, I don't have access to the namespace, nor localName, which are otherwise available in the annotations!